src/crimson/os/seastore/random_block_manager/avlallocator.cc

```diff
@@ -72,19 +72,30 @@ void AvlAllocator::_remove_from_tree(rbm_abs_addr start, rbm_abs_addr size)
   }
 }
 
-rbm_abs_addr AvlAllocator::find_block(size_t size)
+extent_len_t AvlAllocator::find_block(
+  size_t size,
+  rbm_abs_addr &start)
 {
+  uint64_t max_size = 0;
+  auto p = extent_size_tree.rbegin();
+  if (p != extent_size_tree.rend()) {
+    max_size = p->end - p->start;
+  }
+
+  assert(max_size);
+  if (max_size <= size) {
+    start = p->start;
```

Comment:
This looks like the first-fit policy, right? Which one is better here (best-fit vs first-fit)?

Reply:
I'm not sure, @cyx1231st what do you think?

Reply:
My understanding is that currently only object-data-blocks support maybe-alloc-multiple, other kinds of blocks still require the original alloc-single interface, otherwise they will be failed. So I believe we should not drop the original alloc_single_extent in this case.

Reply:
> My understanding is that currently only object-data-blocks support maybe-alloc-multiple, other kinds of blocks still require the original alloc-single interface, otherwise they will be failed. So I believe we should not drop the original alloc_single_extent in this case.

I reserved the single alloc interface at the Cache and TransactionManager layer, so that metadata accessses can call it; while, on the other hand, I unified the disk space allocation interface at the RBM layer. The new multi-alloc interface at the RBM layer works exactly the same as the original single-alloc interface when there exists a piece of continuous disk space that can satisfy the alloc request. Is this ok to you? @cyx1231st

Reply:
Looks good to me.

Can this comment be added to BlockRBManager::alloc_extent(size), IIUC:
```
It allocates as few blocks as possible from the size parameter.
```

Reply:
> This looks like the first-fit policy, right? Which one is better here (best-fit vs first-fit)?

 @myoungwon  Looks to me when max_size <= size, `p` is also the best-fit, right?

Reply:
> Looks good to me.
> 
> Can this comment be added to BlockRBManager::alloc_extent(size), IIUC:
> 
> ```
> It allocates as few blocks as possible from the size parameter.
> ```

Yeah, will do.

Reply:
This always allocates the extent size of the first entry in the `extent_size_tree`, right? Therefore, if the size---input argument---is larger than the extent size, it could allocate a smaller block than the size even though there are more suitable blocks. Sorry, this is more like greedy, not first-fit. 

This is not critical for now. But, we can improve this by making this optional in the future.

Reply:
> This always allocates the extent size of the first entry in the `extent_size_tree`, right? Therefore, if the size---input argument---is larger than the extent size, it could allocate a smaller block than the size even though there are more suitable blocks. Sorry, this is more like greedy, not first-fit.

I think that `extent_size_tree` is ordered by size, so if the last element of this tree(`max_size`) is less than the requested size, there will be no suitable blocks, am I right?

Reply:
Ah, sorry, I misunderstood the intention.